miniapps/src/main/java/mx/yellowme/youst/miniapps/utils/HapticFeedbackHelper.kt

```diff
@@ -0,0 +1,28 @@
+package mx.yellowme.youst.miniapps.utils
+
+import android.content.Context
+import android.os.Build
+import android.os.VibrationEffect
+import android.os.Vibrator
+
+object HapticFeedbackHelper {
+
+    private const val NEWER_DEVICES_MS = 3L
+    private const val OLDER_DEVICES_MS = 15L
+
+    fun generateTouchFeedback(context: Context) {
+        val vibrator = context.getSystemService(Context.VIBRATOR_SERVICE) as Vibrator
+
+        if (!vibrator.hasVibrator()) return
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val effect = VibrationEffect.createOneShot(
+                NEWER_DEVICES_MS,
+                VibrationEffect.DEFAULT_AMPLITUDE
+            )
+            vibrator.vibrate(effect)
+        } else {
+            vibrator.vibrate(OLDER_DEVICES_MS)
+        }
```

Comment:
Maybe we should prefer early escape here and evaluate which one is our default behavior:

```
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
    val effect = VibrationEffect.createOneShot(
        NEWER_DEVICES_MS,
        VibrationEffect.DEFAULT_AMPLITUDE
    )
    vibrator.vibrate(effect)
    return
}
 
vibrator.vibrate(OLDER_DEVICES_MS)
```